javascript-client/src/client.ts

```diff
@@ -68,11 +71,36 @@ export default class S3FileFieldClient {
       apiConfig = {},
     }: S3FileFieldClientOptions,
   ) {
+    // Create axios instances for API and direct S3 requests
     this.api = axios.create({
       ...apiConfig,
       // Add a trailing slash
       baseURL: baseUrl.replace(/\/?$/, '/'),
     });
+    this.s3 = axios.create();
+
+    // Attach axios-retry to both axios instances
+    const raxConfig = {
+      // Retry 3 times on requests that return a response (500, etc) before giving up.
+      retry: 3,
```

Comment:
We need to provide a limit here. Given the exponential backoff, at a certain point, retries should probably halt.